src/DynamoRevit/DynamoRevitApp.cs

```diff
@@ -117,6 +165,34 @@ private static void RegisterAdditionalUpdaters(UIControlledApplication applicati
             Updaters.Add(sunUpdater);
         }
 
+        private void SubscribeApplicationEvents()
+        {
+            UIControlledApplication.Idling += OnApplicationIdle;
+
+            proxy = new EventHandlerProxy();
+
+            UIControlledApplication.ViewActivated += proxy.OnApplicationViewActivated;
+            UIControlledApplication.ViewActivating += proxy.OnApplicationViewActivating;
+
+            ControlledApplication.DocumentClosing += proxy.OnApplicationDocumentClosing;
+            ControlledApplication.DocumentClosed += proxy.OnApplicationDocumentClosed;
+            ControlledApplication.DocumentOpened += proxy.OnApplicationDocumentOpened;
+        }
+
+        private void UnsubscribeApplicationEvents()
+        {
+            UIControlledApplication.Idling -= OnApplicationIdle;
+
+            UIControlledApplication.ViewActivated -= proxy.OnApplicationViewActivated;
+            UIControlledApplication.ViewActivating -= proxy.OnApplicationViewActivating;
+
+            ControlledApplication.DocumentClosing -= proxy.OnApplicationDocumentClosing;
+            ControlledApplication.DocumentClosed -= proxy.OnApplicationDocumentClosed;
+            ControlledApplication.DocumentOpened -= proxy.OnApplicationDocumentOpened;
+
+            proxy = null;
```

Comment:
As `proxy`'s lifetime is as long as application, not need to set it to null. 

Question: why adding event handler is not necessary in onIdle()?